Additions/XCTestCase-KIFAdditions.m

```diff
@@ -65,6 +65,17 @@ - (void)KIF_recordUnexpectedFailureWithDescription:(id)arg1 exception:(NSExcepti
 - (void)writeScreenshotForException:(NSException *)exception;
 {
     [[UIApplication sharedApplication] writeScreenshotForLine:[exception.userInfo[@"LineNumberKey"] unsignedIntegerValue] inFile:exception.userInfo[@"FilenameKey"] description:nil error:NULL];
+    
+    dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
```

Comment:
Is `XCTContext runActivityNamed:block:` not synchronous? Nothing in the documentation seems to indicate that it will invoke the block asynchronously:
https://developer.apple.com/documentation/xctest/xctcontext/2887132-runactivitynamed

Reply:
I was getting crash sometimes that It couldn't get the screenshot and . I fixed it in this way

Reply:
Interesting, I wonder why that would be. If it isn't a synchronous block invocation, then this makes sense to keep it as is. Please add a comment with some information about what the behavior was you were seeing without the semaphore for the next person that encounters this code.